app/oauth.go

```diff
@@ -717,6 +721,15 @@ func (a *App) AuthorizeOAuthUser(w http.ResponseWriter, r *http.Request, service
 	if resp, err := a.HTTPClient(true).Do(req); err != nil {
 		return nil, "", stateProps, model.NewAppError("AuthorizeOAuthUser", "api.user.authorize_oauth_user.service.app_error", map[string]interface{}{"Service": service}, err.Error(), http.StatusInternalServerError)
 	} else {
+		bodyBytes, _ = ioutil.ReadAll(resp.Body)
+		if resp.StatusCode != http.StatusOK {
+			bodyString := string(bodyBytes)
+			mlog.Error("Error getting OAuth user: " + bodyString)
+			if service == model.SERVICE_GITLAB && resp.StatusCode == http.StatusForbidden && strings.Contains(bodyString, "Terms of Service") {
```

Comment:
Kind of ugly, I know, but they don't give us much to work with when checking the error they send us